Sources/Datasets/datasets/structured/Iris.swift

```diff
@@ -0,0 +1,147 @@
+///
+/// This is perhaps the best known database to be found in the pattern recognition
+/// literature. Fisher's paper is a classic in the field and is referenced
+/// frequently to this day. (See Duda & Hart, for example.) The data set contains
+/// 3 classes of 50 instances each, where each class refers to a type of iris
+/// plant. One class is linearly separable from the other 2; the latter are NOT
+/// linearly separable from each other.
+/// 
+/// BibTeX citation:
+/// @misc{Dua:2019,
+///     author = "Dua, Dheeru and Graff, Casey",
+///     year = "2017",
+///     title = "{UCI} Machine Learning Repository",
+///     url = "http://archive.ics.uci.edu/ml",
+///     institution = "University of California, Irvine, School of Information and Computer Sciences"
+/// }
+
+import Foundation
+import S5TF
+import TensorFlow
+
+public struct Iris: S5TFDataset {
+    typealias DataLoader = IrisDataLoader
+    public static var train: IrisDataLoader {
+        get {
+            return IrisDataLoader()
+        }
+    }
+
+    public static var info = irisInfo
+    public static let numberOfTrainingExamples = 150
+    public static let numberOfFeatures = 4
+
+    private init() {}
+}
+
+fileprivate let irisInfo = S5TFDatasetInfo(
+    name: "Iris",
+    version: "0.0.1",
+    description: """
+    This is perhaps the best known database to be found in the pattern recognition
+    literature. Fisher's paper is a classic in the field and is referenced
+    frequently to this day. (See Duda & Hart, for example.) The data set contains
+    3 classes of 50 instances each, where each class refers to a type of iris
+    plant. One class is linearly separable from the other 2; the latter are NOT
+    linearly separable from each other.
+    """,
+    homepage: URL(string: "http://archive.ics.uci.edu/ml")!
+)
+
+public struct IrisDataLoader: S5TFDataLoader {
+    private var index = 0
+
+    let batchSize: Int
+    let data: Tensor<Float>
+    let labels: Tensor<Int32>
+
+    // MARK: - Initializers.
+    private init(batchSize: Int) {
+        self.batchSize = batchSize
+
+        // Download necessary files.
+        let semaphore = DispatchSemaphore(value: 0)
+        var csvUrl: URL? = nil
+        print("Fetching files... Waiting for the download to finish before continuing...")
+        Downloader().download(fileAt: URL(string: "https://archive.ics.uci.edu/ml/machine-learning-databases/iris/iris.data")!,
+            cacheName: "iris", fileName: "iris.csv") { url, err in
+            guard let url = url else {
+                fatalError("URL not saved.")
+            }
+            csvUrl = url
+            semaphore.signal()
+        }
+        semaphore.wait()
+
+        // Load data.
+        let rawData = try! String(contentsOfFile: csvUrl!.absoluteString)
+        // Use a flattened array because Swift does not support 2 dimensional arrays for initialization.
+        var XValues = [Float]()
+        var yValues = [Int32]()
```

Comment:
Could we please be more specific with this variable names, ie `featureValues` and `labelValues`?

Reply:
Good idea!